src/rpc/misc.cpp

```diff
@@ -40,6 +40,52 @@ class DescribeAddressVisitor : public boost::static_visitor<UniValue>
 
     explicit DescribeAddressVisitor(CWallet *_pwallet) : pwallet(_pwallet) {}
 
+    void ProcessSubScript(const CScript& subscript, UniValue& obj, bool include_addresses = false) const
+    {
+        // Always present: script type and redeemscript
+        txnouttype which_type;
+        std::vector<std::vector<unsigned char>> solutions_data;
+        Solver(subscript, which_type, solutions_data);
+        obj.pushKV("script", GetTxnOutputType(which_type));
+        obj.pushKV("hex", HexStr(subscript.begin(), subscript.end()));
+
+        // Only when the script corresponds to an address:
+        CTxDestination embedded;
+        UniValue a(UniValue::VARR);
+        if (ExtractDestination(subscript, embedded)) {
+            UniValue subobj = boost::apply_visitor(*this, embedded);
+            subobj.pushKV("address", EncodeDestination(embedded));
+            subobj.pushKV("scriptPubKey", HexStr(subscript.begin(), subscript.end()));
+            if (subobj.exists("pubkey")) {
+                // Always report the pubkey at the top level, so that `getnewaddress()['pubkey']` always works.
+                obj.pushKV("pubkey", subobj["pubkey"]);
+            }
+            obj.pushKV("embedded", std::move(subobj));
+            a.push_back(EncodeDestination(embedded));
+        }
+
+        // Only when the script is multisig (in which case the ExtractDestination above certainly failed):
+        if (which_type == TX_MULTISIG) {
+            // TODO: abstract out the common functionality between this logic and ExtractDestinations.
+            obj.pushKV("sigsrequired", solutions_data[0][0]);
+            UniValue pubkeys(UniValue::VARR);
+            for (size_t i = 1; i < solutions_data.size() - 1; ++i) {
+                CPubKey key(solutions_data[i].begin(), solutions_data[i].end());
+                a.push_back(EncodeDestination(key.GetID()));
```

Comment:
Same as above.

Reply:
Why?

Reply:
Because `a` is only used if `include_addresses`? Or am I missing something?

Reply:
Thanks, I needed more context to see what you meant. Fixed.